### PR TITLE
Add interactive image canvas with direct manipulation controls

### PR DESCRIPTION
**Pull request by Noah231515**
## Summary
Introduces `ImageCanvasComponent`, a new interactive image viewer with direct manipulation controls (corner resize handles, pan, wheel zoom, double-click fit) that replaces the previous zoom-button-only approach. The component is integrated into the receipt form's inline image carousel while keeping the fullscreen dialog unchanged.

### New Component: `app-image-canvas`
- **`image-canvas.component.ts`**: Standalone component implementing a fixed-viewport image canvas with:
  - Four corner resize handles (aspect-locked, scaling about the opposite corner)
  - Pan via pointer drag (with pointer capture for smooth 120Hz tracking)
  - Wheel zoom anchored at cursor position
  - Double-click to fit image to stage
  - Clamping logic: axes narrower than stage are centered; larger axes are edge-clamped
  - **Single transform on single element** (translate + scale) to avoid drag desynchronization
  - Imperative DOM updates (no template reads viewport) to avoid change detection on every pointermove
  - Native pointer event listeners (not host bindings) for the same reason
- **`image-canvas.component.html`**: Template with image, frame overlay, and four corner handles
- **`image-canvas.component.scss`**: Styling with fixed viewport, handle positioning (fully inside image box, not centered on corners), and cursor feedback
- **`image-canvas.component.spec.ts`**: 268 lines of comprehensive tests covering resize, pan, zoom, clamping, and edge cases

### Integration into Receipt Form
- **`carousel.component.ts`**: Added `directManipulation` and `stageHeight` inputs (opt-in, following `hideButtonControls` precedent)
- **`carousel.component.html`**: Pass new inputs to `app-image-viewer`
- **`carousel.component.scss`**: New `.rw-carousel--fill` class to make carousel fill its column; shrinks ngx-bootstrap's prev/next controls to centered buttons clear of corners
- **`image-viewer.component.ts`**: Added `directManipulation` and `stageHeight` inputs; skip wheel re-emit when canvas owns it
- **`image-viewer.component.html`**: Conditional render: canvas when `directManipulation()` is true, else original scaled image
- **`receipt-form.component.ts`**: 
  - Changed `carouselComponent` from `viewChild.required` to `viewChild` (carousel not rendered when images hidden)
  - Removed `showLargeImagePreview` boolean; replaced with `currentImageIndex` getter
  - Removed `expandedImageDialog` field initialization (now declared inline)
  - Pass `directManipulation=true` and `stageHeight="100%"` to inline carousel only
- **`receipt-form.component.html`**: 
  - Details pane: `col` → `col-12 col-md` (responsive stacking)
  - Images pane: Added `rw-images-pane` class; restructured with `rw-images-pane__content` and `rw-images-pane__stage` divs
  - Carousel: Added `rw-carousel--fill` class and `stageHeight="100%"`
- **`receipt-form.component.scss`**: New height chain for inline image stage (flex layout, responsive 50vh fallback below md breakpoint); fullscreen dialog kept unchanged via class gating

### Removed User Preference
- **`user-preferences.component.html`**: Removed "Default to show large image previews?" checkbox section
- **`user-preferences.component.ts`**: Removed `showLargeImagePreviews` form control
- **`user-preferences.component.spec.ts`**: Removed related test setup

### Module Registration
- **`shared-ui.module.ts`**: Added `ImageCanvasComponent` to imports and exports

## Notable Implementation Details

- **No change detection on drag**: Viewport is a plain field (not a signal); image and frame positioned imperatively via `Renderer2.setStyle()`. Pointer events are native listeners, not host bindings. This keeps a 120Hz drag from triggering change detection on the receipt form's large template.
- **Handles fully

https://claude.ai/code/session_01N5YQ81YHYVwddND4FjXWpj

## Summary by CodeRabbit

* **New Features**
  * Added interactive receipt image viewing with panning, corner resizing, zoom controls, double-click fitting, and responsive sizing.
  * Added fullscreen image viewing with a close button while preserving the selected image.
  * Improved carousel presentation with contained controls and fill-to-height layouts.
* **Bug Fixes**
  * Image actions and zoom controls now continue working when the carousel is not rendered.
* **Changes**
  * Removed the receipt image preview-size toggle and its associated preference.

### CODE REVIEW

File: desktop/CLAUDE.md [hunk 1]
```diff
@@ -1638,6 +1638,127 @@ Two cross-component seams support this (each with its own focused spec):
...
+- Wheel zoom is anchored at the cursor and moves a meaningful amount. The viewer this replaced
+  multiplied `deltaY` by `-0.000001`, so one notch changed the scale by 0.0001 — wheel zoom did
+  nothing at all.
+
+**Wiring, and how fullscreen stays untouched.** `app-carousel` gained `directManipulation` +
+`stageHeight`, passed to `app-image-viewer`, which renders the canvas when the flag is on and its
+original markup when off — the `hideButtonControls` precedent. The receipt form passes the flag on
+the **inline** carousel and deliberately not on `#expandedImageTemplate`, so the fullscreen dialog
+keeps the default and is byte-identical. `ImageViewerComponent.onWheel` also stops re-emitting to the
+carousel in canvas mode, or the wheel would drive the carousel's shared scale as well.
```
[the remark this answers is not in the document]
Comment (Noah231515) [reply desktop/CLAUDE.md]: **ACCEPT** for the first site; the second was already fixed.

Verified, and it was worse than reported. The paragraph was still headed *"Wiring, and how fullscreen stays untouched"* and claimed the flag was passed *"deliberately not on `#expandedImageTemplate`, so the fullscreen dialog … is byte-identical"* — while the paragraph ~45 lines below it already said *"The fullscreen dialog runs the same canvas."* The file contradicted itself; the fullscreen port added the correct text and left the old text in place.

Rewritten so there is one account: both carousels get the canvas, there is no opt-out, and the `directManipulation` flag that used to select between two rendering paths is gone along with the old path.

The second site (then 1749-1750) was already corrected in `5f90da8`, before this review — it now reads that both carousels carry the fill class and `stageHeight="100%"`, which matches what the spec asserts.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: desktop/src/carousel/carousel/carousel.component.ts [hunk 2]
```diff
@@ -42,13 +54,31 @@ export class CarouselComponent implements OnChanges {
...
     this.adjustScale(0.1);
   }
 
+  /**
+   * On a canvas each image owns its own zoom and pan, so the header buttons act
+   * on the slide being looked at rather than on one scale shared by all of them.
+   */
+  private activeViewer(): ImageViewerComponent | undefined {
+    return this.viewers()[this.currentlyShownImageIndex];
+  }
```
[the remark this answers is not in the document]
Comment (Noah231515) [reply desktop/src/carousel/carousel/carousel.component.ts:66]: **ACCEPT**, with a smaller fix than either option suggested — and one correction to the diagnosis.

The mechanism is real: `<slide>` is ungated in both loops while `app-image-viewer` was `*ngIf`-gated, so `viewers()[currentlyShownImageIndex]` was a **slide** index into a **viewer** array.

Two things the investigation turned up:

1. **No reachable input state triggers it today.** The preview loop binds `preview?.file`, which is non-optional on `ReceiptFileUploadCommand` and always set at construction in `upload-image.component.ts`; the images loop binds `encodedImage`, which is required on `FileDataView` and whose failed fetches are dropped from the array entirely (`receipt-form.component.ts:601-605`). Both guards were effectively `*ngIf="true"`. So this was latent fragility, not a live defect — worth fixing, but it was not breaking zoom.

2. **The cause named in the comment isn't the one at work.** `image-viewer`'s own outer `@if` lives inside that component's template, so it can never remove the component from the parent's `viewChildren` query. Only the carousel-side `*ngIf` gated instantiation.

That second point is what makes the cheap fix correct: since the viewer already renders nothing without a source, the carousel's `*ngIf`s were **redundant**. Deleting them makes `viewers()` 1:1 with slides *by construction* — a stronger guarantee than an explicit mapping, and two fewer lines rather than a new structure to keep in sync. Comments on both loops say not to reintroduce them, and `activeViewer()`'s doc comment now states the invariant it depends on.

This also matters more than it did: the same commit removes the dead pre-canvas viewer, so `zoomIn`/`zoomOut` no longer have a non-`activeViewer()` branch to fall back to.

Verified in the browser on a 5-image receipt: 5 slides / 5 viewers / 5 canvases, and clicking header Zoom In twice changed the width of **only** the active slide's image.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: desktop/src/receipts/receipt-form/receipt-form.component.spec.ts [hunk 3]
```diff
@@ -1205,4 +1205,56 @@ describe("ReceiptFormComponent", () => {
...
+    // Without this the dialog offers no way out but Esc and the backdrop.
+    it("closes the fullscreen dialog from its close button", async () => {
+      const dialog = TestBed.inject(MatDialog);
+      openDialogCarousel();
+      expect(dialog.openDialogs.length).toEqual(1);
+
+      component.closeExpandedImage();
+      await fixture.whenStable();
+
+      expect(dialog.openDialogs.length).toEqual(0);
```
[the remark this answers is not in the document]
Comment (Noah231515) [reply desktop/src/receipts/receipt-form/receipt-form.component.spec.ts:1258]: **ACCEPT** — the gap was real, though the suggested mechanism doesn't work in this spec.

The test was named *"from its close button"* and called `closeExpandedImage()` directly, so it passed with the `(clicked)` binding — or the whole button — deleted. Confirmed by removing the binding: the old test still passed.

Querying the rendered native button isn't possible here. This suite runs with `CUSTOM_ELEMENTS_SCHEMA` and declares only `ReceiptFormComponent`, so `app-button` is an unknown element: no inner `<button>` is rendered and the `data-testid` sits on the `app-button` host, which doesn't forward it. Importing the real `ButtonComponent` and its Material deps into a 51-test suite to reach one button is a poor trade.

What that configuration *does* give is the binding as a plain DOM listener, which is exactly what was untested. Split into two tests:

- `closes the fullscreen dialog` — drives the method, asserts `openDialogs` empties (unchanged behaviour coverage).
- `wires the close button to closeExpandedImage` — dispatches the `clicked` event on the button element, asserts the handler ran.

The second is mocked with `mockImplementation` because actually closing from a DOM event runs a change-detection pass this block doesn't stub view children for.

Verified it bites: with `(clicked)` removed, the new test fails with *"Expected number of calls: >= 1, Received: 0"*.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: desktop/src/shared-ui/image-canvas/image-canvas.component.html [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    [src]="src()"
+    [alt]="alt()"
+    (load)="onImageLoad()"
+  />
+  <div #frame class="rw-image-canvas__frame">
+    @for (corner of corners; track corner) {
+      <span
+        class="rw-image-canvas__handle"
+        [attr.data-corner]="corner"
+        (pointerdown)="onResizeStart($event, corner)"
```
[the remark this answers is not in the document]
Comment (Noah231515) [reply desktop/src/shared-ui/image-canvas/image-canvas.component.html:15]: **ACCEPT** — this was a genuine bug, and the best find in the review. Applied the suggestion, plus a test.

Confirmed rather than assumed. The handle `<span>` is a descendant of the host, the host binds `(dblclick)="fit()"`, and `onResizeStart`'s `preventDefault()` / `stopPropagation()` act on the **PointerEvent** — `dblclick` is a separate event and neither call suppresses it. So quickly pulling a corner twice discarded the size just dragged to.

Written as a failing test first, in `image-canvas.component.spec.ts`:

```
- "scale": 0.4166666666666667,   <- after the corner drag
+ "scale": 0.25,                 <- after double-clicking the handle: back to fit
```

With `(dblclick)="$event.stopPropagation()"` on the handle it passes, and the existing test asserting that double-clicking the **host** still fits is untouched — the two together pin both halves.

Also verified in the browser on a real receipt: corner drag takes the image from 383px to 532px wide, and double-clicking the handle leaves it at 532px, while double-clicking the image itself still returns it to the fit.

Recorded in `desktop/CLAUDE.md` beside the other canvas rules, since the pointer-vs-mouse event distinction is the kind of thing that gets "simplified" away later.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: desktop/src/shared-ui/image-canvas/image-canvas.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+  it("zooms in and back out again", async () => {
+    component.zoomIn();
+    await fixture.whenStable();
+    const zoomed = viewport().scale;
+    expect(zoomed).toBeGreaterThan(FIT_SCALE);
+
+    component.zoomOut();
+    await fixture.whenStable();
+    expect(viewport().scale).toBeCloseTo(FIT_SCALE);
+  });
```
[the remark this answers is not in the document]
Comment (Noah231515) [reply desktop/src/shared-ui/image-canvas/image-canvas.component.spec.ts:214]: **ACCEPT** — added, essentially as suggested.

Verified the gap is real: `clampScale` floors at `fitScale()` and ceils at `MAX_SCALE`, the floor has a test (`"will not zoom out past the fit"`), the ceiling had none. The most `zoomIn()` calls anywhere in the suite was 2, from a `FIT_SCALE` of 0.25 — roughly 25 are needed to reach 8, so the cap was never approached.

Placed next to the floor test so the pair reads together, with a local `MAX_SCALE` constant rather than a bare `8` (the component doesn't export its own; the constant names the source, matching how `FIT_SCALE` already mirrors the component's formula).

Verified it bites: dropping `Math.min(..., MAX_SCALE)` from `clampScale` makes it fail, and restoring it passes.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: api/internal/models/user_preferences_test.go [hunk 6]
```diff
@@ -8,7 +8,7 @@ import (
 )
 
 func TestUserPrefernces_LoadDataFromRequest(t *testing.T) {
-	body := `{"userId": 5, "showLargeImagePreviews": true, "quickScanDefaultStatus": "OPEN"}`
+	body := `{"userId": 5, "quickScanDefaultStatus": "OPEN"}`
```
[the remark this answers is not in the document]
Comment (Noah231515) [reply api/internal/models/user_preferences_test.go:11]: Both findings on this thread: **ACCEPT** for each.

**Run the required API validation.** Done, on the current branch:

```
go test -count=1 ./...                                    -> exit 0, all packages pass
go test -coverprofile=... -covermode=atomic ./...         -> total: 62.0% of statements
```

For the code this change touches:

| Scope | Coverage |
|---|---|
| `internal/models` (package) | **96.0%** |
| `UserPrefernces.LoadDataFromRequest` | **100.0%** |

(`coverage.out` is a build artifact and isn't committed.)

**Keep a legacy-payload regression case.** Good catch, and accepted for a stronger reason than stated: it pins the compatibility argument the whole removal rests on. The API sets `DisallowUnknownFields` nowhere, so an already-released client still sending `showLargeImagePreviews` gets a 200 with the field ignored rather than a 400 — that, plus the generated Dart field having been `bool?`, is what made dropping the field safe for shipped mobile builds.

Implemented as a **dedicated test** rather than by restoring the key to the existing fixture:

```go
func TestUserPrefernces_LoadDataFromRequestIgnoresRemovedFields(t *testing.T)
```

Restoring a removed field to the primary test's payload with no explanation reads like leftover debris, and the next person to touch the file removes it — taking the coverage with it. A named test with a comment saying *"Keep the removed field in this payload - it is the whole point of the test"* can't be misread, and leaves the primary test's fixture describing only current fields.

Verified against the running API as well: `PUT /api/userPreferences` still carrying the field returns 200, ignores it, and persists the remaining fields.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: desktop/src/shared-ui/image-canvas/image-canvas.component.ts [hunk 7]
```diff
@@ -0,0 +1,397 @@
...
+  protected onWheel(event: WheelEvent): void {
+    if (!this.natural.width) {
+      return;
+    }
+
+    event.preventDefault();
+
+    const rect = this.host.nativeElement.getBoundingClientRect();
+    this.zoomAround(
+      event.deltaY < 0 ? ZOOM_STEP : 1 / ZOOM_STEP,
```
[the remark this answers is not in the document]
Comment (Noah231515) [reply desktop/src/shared-ui/image-canvas/image-canvas.component.ts:165]: **ACCEPT** — real bug, fixed with the regression test you asked for.

`onWheel` read `event.deltaY < 0 ? ZOOM_STEP : 1 / ZOOM_STEP`, so `deltaY === 0` took the else branch and zoomed **out**. A two-finger horizontal trackpad swipe over the image shrank it, and the unconditional `preventDefault()` above swallowed the horizontal scroll on the way past.

The guard returns **before** `preventDefault()`, so the event keeps its default behaviour rather than just skipping the zoom.

Writing the test first turned out to matter, because the obvious version passes on the broken code. Dispatching a `deltaY: 0` wheel at the fit scale leaves the viewport unchanged either way — `clampScale` floors at `fitScale()`, so the spurious zoom-out is clamped to a no-op. The test therefore zooms in twice first, which makes the bug visible:

```
-   "scale": 0.33062499999999995,   <- expected (unchanged)
+   "scale": 0.2875,                <- actual: the horizontal swipe zoomed out
```

It asserts both halves — the viewport is unchanged **and** `event.defaultPrevented` is false — since a fix that stopped zooming but still ate the scroll would otherwise pass. Verified to fail with the guard reverted and pass with it applied.

Also checked in a real browser, as jsdom does not model trackpads: a vertical wheel still zooms in (383px → 440px) and out (440px → 383px), while a horizontal swipe now leaves the size at 440px.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: desktop/CLAUDE.md [hunk 8]
```diff
@@ -1731,6 +1731,152 @@ Two cross-component seams support this (each with its own focused spec):
...
+the carousel chain resolves its `100%` against that. Leave that media query out and the stacked
+canvas collapses to its 2px border.
+
+**`app-upload-image` must not carry `h-100` in that column.** The class was inert while its host was
+`display: inline`, but a flex container blockifies its children — at which point
+`height: 100% !important` makes a hidden file input swallow the whole column and leaves the stage
+with nothing but its own 2px border.
+
+**The fullscreen dialog runs the same canvas.** `#expandedImageTemplate` now passes
+`class="rw-carousel--fill"`, `[directManipulation]="true"` and `stageHeight="100%"` exactly as the
```
[the remark this answers is not in the document]
Comment (Noah231515) [reply desktop/CLAUDE.md]: **ACCEPT** — correct, and this one is my own regression.

The commit that removed the dead pre-canvas viewer deleted the `directManipulation` input from both `CarouselComponent` and `ImageViewerComponent` and dropped the binding from both call sites, but this paragraph still listed it among the three things `#expandedImageTemplate` passes. Two of those three are real; the middle one named a binding that no longer exists.

Removed, leaving `class="rw-carousel--fill"` and `stageHeight="100%"`.

Worth noting this is the second stale-docs finding of the same shape on this PR — fixing one paragraph and leaving its twin behind — so I checked for siblings rather than just the flagged line. `directManipulation` survives at one other place (line 1773), but correctly: that sentence describes the flag as something that *was* deleted. The mentions of `img.viewer-image`, `scale` / `adjustScale()` / `onScroll()` and the `[scale]` / `wheel` plumbing are likewise all in the past tense, describing what the cleanup removed. So 1831 was the only stale claim.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: desktop/src/shared-ui/shared-ui.module.ts [hunk 9]
```diff
@@ -141,6 +142,7 @@ import { BadgeComponent } from './badge/badge.component';
     PieChartUiComponent,
     LoginQrComponent,
     BadgeComponent,
+    ImageCanvasComponent,
```
[the remark this answers is not in the document]
Comment (Noah231515) [reply desktop/src/shared-ui/shared-ui.module.ts:145]: **REJECT** — false positive. `ImageCanvasComponent` is already in `imports`, not `declarations`.

`shared-ui.module.ts` has three arrays, and line 145 falls in the second one:

| Array | Starts at |
|---|---|
| `declarations:` | line 73 |
| `imports:` | line **116** |
| `exports:` | line 147 |

So line 145 is inside **`imports`** — which is exactly where a standalone component belongs — and the second reference at line 191 is inside `exports`, which is the arrangement the comment asks for. Nothing to change.

The independent proof is that this cannot be a latent defect: Angular rejects a standalone component in `declarations` **at compile time**, and both `ng build` and the full Jest suite (1682 tests, 246 suites) pass on this branch. A 🔴 Critical defect of this kind would take the build down rather than sit unnoticed.

I suspect the analysis script is what misled it here — it printed `sed -n '60,80p;135,200p'`, which skips lines 81-134 and so cuts out the `imports:` opener at 116. Line 145 then looks like it belongs to the last array header visible in the output.

---
_Generated by [Claude Code](https://claude.ai/code)_
